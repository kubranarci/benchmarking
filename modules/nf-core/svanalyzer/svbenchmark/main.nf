
process SVANALYZER_SVBENCHMARK {
    tag "$meta.id $meta2.caller"
    label 'process_medium'

    conda "bioconda::svanalyzer=0.35"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/svanalyzer:0.35--pl526_0':
        'quay.io/biocontainers/svanalyzer:0.35--pl526_0' }"

    input:
    tuple val(meta),val(meta2), path(test), path(test_index), path(truth), path(truth_index)
    tuple path(fasta), path(fai)
    path(bed)

    output:
    tuple val(meta), path("*.falsenegatives.vcf"), emit: fns
    tuple val(meta), path("*.falsepositives.vcf"), emit: fps
    tuple val(meta), path("*.distances")         , emit: distances
    tuple val(meta), path("*.log")               , emit: log
    tuple val(meta), path("*.report")            , emit: report
    path "versions.yml"                          , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def bed = bed ? "-includebed $bed" : ""
    def VERSION = '0.35' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.

    """
    svanalyzer \\
        benchmark \\
        $args \\
        --ref $fasta \\
        --test $test \\
        --truth $truth \\
        --prefix $prefix \\
        $bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        svanalyzer: ${VERSION}
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.35' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.

    """
    touch ${prefix}.falsenegatives.vcf
    touch ${prefix}.falsepositives.vcf
    touch ${prefix}.distances
    touch ${prefix}.log
    touch ${prefix}.report

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        svanalyzer: ${VERSION}
    END_VERSIONS
    """
}
