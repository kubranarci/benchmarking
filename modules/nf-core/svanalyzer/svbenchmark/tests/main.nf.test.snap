{
    "homo_sapiens - illumina - vcf.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.falsenegatives.vcf:md5,7a574ad3c614f0c38a6dccb5901ddc05"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.falsepositives.vcf:md5,146f05125330a9c0c4581a5a581744a7"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.distances:md5,b6a908b1187f159429dc8c6beb6e6f57"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,16fd250ae2adcb07c8f101f640fdce81"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test.report:md5,d0c7db2466442cbdd4f87ad516e00d32"
                    ]
                ],
                "5": [
                    "versions.yml:md5,6ac08491dbe591d24615c131658cf771"
                ],
                "distances": [
                    [
                        {
                            "id": "test"
                        },
                        "test.distances:md5,b6a908b1187f159429dc8c6beb6e6f57"
                    ]
                ],
                "fns": [
                    [
                        {
                            "id": "test"
                        },
                        "test.falsenegatives.vcf:md5,7a574ad3c614f0c38a6dccb5901ddc05"
                    ]
                ],
                "fps": [
                    [
                        {
                            "id": "test"
                        },
                        "test.falsepositives.vcf:md5,146f05125330a9c0c4581a5a581744a7"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,16fd250ae2adcb07c8f101f640fdce81"
                    ]
                ],
                "report": [
                    [
                        {
                            "id": "test"
                        },
                        "test.report:md5,d0c7db2466442cbdd4f87ad516e00d32"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6ac08491dbe591d24615c131658cf771"
                ]
            }
        ],
        "timestamp": "2023-11-24T15:28:29.422156227"
    },
    "homo_sapiens - illumina - vcf.gz - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.falsenegatives.vcf:md5,c04105833d5f9acb5ec50af0b89a6b0f"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.falsepositives.vcf:md5,146f05125330a9c0c4581a5a581744a7"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.distances:md5,95f7b1679e805979f0c9a4a322cb77f4"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,75f05aac48afd818f2d5e8a3144116ea"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test.report:md5,aaa9225dd6e261951c020e23473375b9"
                    ]
                ],
                "5": [
                    "versions.yml:md5,6ac08491dbe591d24615c131658cf771"
                ],
                "distances": [
                    [
                        {
                            "id": "test"
                        },
                        "test.distances:md5,95f7b1679e805979f0c9a4a322cb77f4"
                    ]
                ],
                "fns": [
                    [
                        {
                            "id": "test"
                        },
                        "test.falsenegatives.vcf:md5,c04105833d5f9acb5ec50af0b89a6b0f"
                    ]
                ],
                "fps": [
                    [
                        {
                            "id": "test"
                        },
                        "test.falsepositives.vcf:md5,146f05125330a9c0c4581a5a581744a7"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,75f05aac48afd818f2d5e8a3144116ea"
                    ]
                ],
                "report": [
                    [
                        {
                            "id": "test"
                        },
                        "test.report:md5,aaa9225dd6e261951c020e23473375b9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6ac08491dbe591d24615c131658cf771"
                ]
            }
        ],
        "timestamp": "2023-11-24T15:29:05.547215282"
    }
}